os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -244,104 +247,64 @@ static void reload_kill_each(FAR struct tcb_s *tcb, FAR void *arg)
 	}
 }
 
-static int reload_kill_binary(int binid)
+int reload_kill_binary(int binid)
 {
 	int ret;
 
 	if (binid < 0) {
 		return ERROR;
 	}
 
+	sched_lock();
```

Comment:
There is sched_lock and sched_unlock in the task_terminate() already.
Is it needed for sched_foreach?

Reply:
This function touches all tcb list with sched_foreach but loading thread is a task with the highest priority.
It can bring any problems in the middle of operation so it needs something to lock scheduling I think.

Reply:
This function touches all tcb list with sched_foreach but loading thread is a task with the highest priority.
It can bring any problems in the middle of operation so it needs something to lock scheduling I think.